### DISCUSSION TITLE
get qtile to recognize screen size if it is set in startup

### DISCUSSION THREAD

**Discussion by ExpandingMan** [qtile/qtile#4951] [Q&A]
I had been setting my resolution in a bash script that is called by qtile on startup (in X11).  After a recent update (0.27 I guess? thought it was more recent, whatever), qtile does not seem to set its screen size correctly after this, as evidenced by incorrect bar position.  Restarting qtile in place allows it to recognize the screen size just fine and I can use it as normal.

What's the recommended way to fix this?  Also open to better ways of setting the resolution if there are any suggestions.

**Comment by ExpandingMan** [comment]
I have discovered that `reconfigure_screens` works in principle, but I am unable to figure out where to put it in the initialization process so that it correctly resets the screens.  If I put it in the function call that starts up, it doesn't do anything.  Same if I put it as a CLI command in the startup bash script at the very end.  After everything is all loaded, executing it manually fixes the problem.

Also I just realized there is the config option `reconfigure_screens` which I already had set to `True`.  Setting it to `False` does not seem to have made any difference.

**Comment by tych0** [comment]
I believe this is a dupe of #4936

**Comment by ExpandingMan** [comment]
Hm, probably yes.

**Comment by cr1ogen** [comment]
There were once problems with the bar, but in wayland. From what you say, it might be similar to this. https://github.com/qtile/qtile/issues/3181 because the same thing happens to me when I enter x11 these days

**Comment by tych0** [comment]
I'm gonna go ahead and close this, since we reverted the buggy commit and have released 0.28.0, let me know if that doesn't fix it.
